core/src/avm1/object/stage_object.rs

```diff
@@ -177,29 +185,38 @@ impl<'gc> TObject<'gc> for StageObject<'gc> {
         let name = name.into();
         let obj = self.0.read();
         let props = activation.context.avm1.display_properties;
-        let case_sensitive = activation.is_case_sensitive();
+
         // Property search order for DisplayObjects:
-        if self.has_own_property(activation, name) {
-            // 1) Actual properties on the underlying object
-            obj.base.get_local_stored(name, activation)
-        } else if let Some(level) =
-            Self::get_level_by_path(name, &mut activation.context, case_sensitive)
-        {
-            // 2) _levelN
-            Some(level)
-        } else if let Some(child) = obj
+        // 1) Actual properties on the underlying object
+        if let Some(value) = obj.base.get_local_stored(name, activation) {
+            return Some(value);
+        }
+
+        // 2) Path properties such as `_root`, `_parent`, `_levelN` (obeys case sensitivity)
+        let magic_property = name.starts_with(b'_');
```

Comment:
I guess this check if for performance only, since it doesn't affect the lookup logic at all.